src/parsec.rs

```diff
@@ -161,14 +236,35 @@ impl ParsecMap {
 
     #[cfg(feature = "mock_base")]
     pub fn has_unpolled_observations(&self) -> bool {
-        let parsec = if let Some(parsec) = self.map.values().last() {
+        let parsec = if let Some(parsec) = self.get_parsec_values().last() {
             parsec
         } else {
             return false;
         };
 
         parsec.has_unpolled_observations()
     }
+
+    fn get_parsec_mut(&mut self, version: u64) -> Option<&mut Parsec> {
+        self.map.get_mut(&version).map(|p| &mut p.parsec)
+    }
+
+    fn get_parsec_values(&self) -> impl Iterator<Item = &Parsec> {
+        self.map.values().map(|p| &p.parsec)
+    }
+
+    fn get_parsec_values_mut(&mut self) -> impl Iterator<Item = &mut Parsec> {
+        self.map.values_mut().map(|p| &mut p.parsec)
+    }
+
+    #[allow(unused)]
```

Comment:
is it best to use `#[cfg(all(test, feature = "mock_parsec"))]`, this way we are not silencing dead code warning

Reply:
Agree good point